no fdatasync prototype on OS X

`fdatasync` is apparently available as a system call in OS X, but there's no prototype for it in `unistd.h`.  

The `configure` test (which only checks that a program using `fdatasync` can be linked) succeeds, but compiling the source reveals the problem:

```
src/unix/jobs-unix/lwt_unix_job_fdatasync.c:61:17: warning: implicit declaration of function 'fdatasync' is invalid in C99 [-Wimplicit-function-declaration]
  job->result = fdatasync(job->fd);
                ^
1 warning generated.
```

--- a/utils/discover.ml
+++ b/utils/discover.ml
@@ -200,7 +200,8 @@ let fdatasync_code = "
 
 CAMLprim value lwt_test(value Unit)
 {
-  fdatasync(0);
+  int (*fdatasyncp)(int) = fdatasync;
+  fdatasyncp(0);
   return Val_unit;
 }
 "

